### PR TITLE
[Issue #19912]-Added unit test for JDBCRepository

### LINKED ISSUE #19912: Add test cases for JDBCRepository
- [ ] Add fixture: JDBCRepositoryFixture implements JDBCRepository
- [ ] Add test case with mock

### PR DESCRIPTION
**Pull request by abhinav-koppula**
Fixes #19912.

Changes proposed in this pull request:
- Added a fixture for JDBCRepositoryProvider
- Covered all positive and negative scenarios for all the methods of JDBCRepository using mocks

### CODE REVIEW

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    public void setup() throws Exception {
+        when(mockJdbcConnection.createStatement()).thenReturn(mockStatement);
+    }
+    
+    @Test
+    public void assertInit() {
+        try (MockedConstruction<HikariDataSource> ignored = mockConstruction(HikariDataSource.class, (mock, context) -> when(mock.getConnection()).thenReturn(mockJdbcConnection))) {
+            JDBCRepository repository = new JDBCRepository();
+            repository.init(getHikariProperties());
+
```
[the remark this answers is not in the document]
Comment (abhinav-koppula) [reply shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java]: @agentart - Thanks for reviewing. Have removed unnecessary blank lines - https://github.com/apache/shardingsphere/pull/20699/commits/06597497baf679008319d4099e571b182ac8322e

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockedConstruction;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JDBCRepositoryTest {
```
Comment (agentart) [inline shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java:49]: Design class as final class except abstract class for extend.

Comment (abhinav-koppula) [reply shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java]: Made JDBCRepositoryTest as final - https://github.com/apache/shardingsphere/pull/20699/commits/382b5b138a1f5cec40ff03f66d1cb43d75aa8a1d

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    private final JDBCRepositoryProviderFixture fixture = new JDBCRepositoryProviderFixture();
+    
+    @Before
+    public void setup() throws Exception {
+        when(mockJdbcConnection.createStatement()).thenReturn(mockStatement);
+    }
+    
+    @Test
+    public void assertInit() {
+        try (MockedConstruction<HikariDataSource> ignored = mockConstruction(HikariDataSource.class, (mock, context) -> when(mock.getConnection()).thenReturn(mockJdbcConnection))) {
```
Comment (agentart) [inline shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java:75]: Would it be better to put duplicate logic in @Before and release resources in @After

Comment (abhinav-koppula) [reply shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-core/src/test/java/org/apache/shardingsphere/mode/repository/standalone/jdbc/JDBCRepositoryTest.java]: Fixed here - https://github.com/apache/shardingsphere/pull/20699/commits/3109141c3541894bc3c2e0583d38797bb2664104

**Review by agentart — APPROVED** [review]

**Pull request merged**